x-pack/plugins/ml/public/shared_components/single_metric_viewer/single_metric_viewer.tsx

```diff
@@ -0,0 +1,252 @@
+  uuid: string;
+}
+
+const SingleMetricViewerWrapper: FC<SingleMetricViewerPropsWithDeps> = ({
+  // Component dependencies
+  coreStart,
+  pluginStart,
+  mlServices,
+  // Component props
+  bounds,
+  functionDescription,
+  lastRefresh,
+  onError,
+  selectedDetectorIndex,
+  selectedEntities,
+  selectedJobId,
+  uuid,
+}) => {
+  const [chartWidth, setChartWidth] = useState<number>(0);
+  const [zoom, setZoom] = useState<AppStateZoom | undefined>();
+  const [selectedForecastId, setSelectedForecastId] = useState<string | undefined>();
+  const [selectedJob, setSelectedJob] = useState<MlJob | undefined>();
+  const [jobsLoaded, setJobsLoaded] = useState(false);
+
+  const isMounted = useRef(true);
+
+  const { mlApiServices, mlJobService, mlTimeSeriesExplorerService, toastNotificationService } =
+    mlServices;
+  const startServices = pick(coreStart, 'analytics', 'i18n', 'theme');
+  const datePickerDeps: DatePickerDependencies = {
+    ...pick(coreStart, ['http', 'notifications', 'theme', 'uiSettings', 'i18n']),
+    data: pluginStart.data,
+    uiSettingsKeys: UI_SETTINGS,
+    showFrozenDataTierChoice: false,
+  };
+
+  const previousRefresh = usePrevious(lastRefresh ?? 0);
+
+  useEffect(
+    function setUpJobsLoaded() {
+      async function loadJobs() {
+        try {
+          await mlJobService.loadJobsWrapper();
+          setJobsLoaded(true);
+        } catch (e) {
+          if (onError) {
+            onError(new Error(errorMessage));
+          }
+        }
+      }
+      if (isMounted.current === false) {
+        return;
+      }
+      loadJobs();
+
+      return () => {
+        isMounted.current = false;
+      };
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+
+  useEffect(
+    function setUpSelectedJob() {
+      async function fetchSelectedJob() {
+        if (mlApiServices && selectedJobId !== undefined) {
+          try {
+            const { jobs } = await mlApiServices.getJobs({ jobId: selectedJobId });
+            const job = jobs[0];
+            setSelectedJob(job);
+          } catch (e) {
+            if (onError) {
+              onError(new Error(errorMessage));
+            }
+          }
+        }
+      }
+      if (isMounted.current === false) {
+        return;
+      }
+      fetchSelectedJob();
+    },
+    [selectedJobId, mlApiServices, onError]
+  );
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const resizeHandler = useCallback(
+    throttle((e: { width: number; height: number }) => {
+      if (Math.abs(chartWidth - e.width) > minElemAndChartDiff) {
+        setChartWidth(e.width);
+      }
+    }, RESIZE_THROTTLE_TIME_MS),
+    [chartWidth]
+  );
+
+  const autoZoomDuration = useMemo(() => {
+    if (!selectedJob) return;
+    return mlTimeSeriesExplorerService?.getAutoZoomDuration(selectedJob);
+  }, [mlTimeSeriesExplorerService, selectedJob]);
+
+  const appStateHandler = useCallback(
+    (action: string, payload?: any) => {
```

Comment:
```suggestion
    (action: string, payload?: AppStateZoom | string) => {
```

Reply:
Updated in 6c61014b8a7ef4a3167a642e5f72f1938ea25c50